src/common/models/Projection.ts

```diff
@@ -491,6 +514,41 @@ export class Projection {
     );
   }
 
+  private calcCaseDensityByCases(): Array<number | null> {
+    const caseDensityByCases = [];
+    const totalPopulation = this.totalPopulation;
+    for (let i = 0; i < this.dates.length; i++) {
+      const weeklyAverage = this.getWeeklyAverageForDay(
+        this.dailyPositiveTests,
+        i,
+      );
+      if (weeklyAverage === null) {
+        caseDensityByCases.push(null);
+      } else {
```

Comment:
nit: Consider adding a check to ensure that `totalPopulation` is greater than `0`. Also, maybe you can simplify the expression

```js
const value = (weeklyAverage && totalPopulation > 0)
  ? weeklyAverage / (totalPopulation / 100000)
  : null;
caseDensityByCases.push(value);
```

(not sure if that's much better TBH)

Reply:
just a nit on Pablo's suggestion: `weeklyAverage` could be `0` so I suspect we do need to do `(weeklyAverage !== null && ...` not just `(weeklyAverage && ...` 😁